test/argsPass.js

```diff
@@ -0,0 +1,155 @@
+        assert.isTrue(p1True.calledOnce);
+        assert.isTrue(p2True.calledOnce);
+        assert.isTrue(p3False.calledOnce);
+        assert.isTrue(c1False.calledOnce);
+        assert.isTrue(c2False.calledOnce);
+      });
+    });
+  });
+
+  context('with more arguments than predicates', function() {
+    context('with all predicates passing', function() {
+      specify(
+        'tests args with supplied predicates and passes remaining args untouched',
+        function() {
+          const f = argsPass(c1True, [p1True, p2True, p3True]);
+          assert.isTrue(f(1, 2, 3, 4));
+          assert.isTrue(p1True.calledWith(1));
+          assert.isTrue(p2True.calledWith(2));
+          assert.isTrue(p3True.calledWith(3));
+          assert.isTrue(c2True.calledWith([true, true, true, 4]));
+          assert.isTrue(p1True.calledOnce);
+          assert.isTrue(p2True.calledOnce);
+          assert.isTrue(p3True.calledOnce);
+          assert.isTrue(c1True.calledOnce);
+          assert.isTrue(c2True.calledOnce);
+        }
+      );
+    });
+
+    context('with one predicate failing', function() {
+      specify('tests args with supplied predicates', function() {
+        const f = argsPass(c1False, [p1True, p2True, p3False]);
+        assert.isFalse(f(1, 2, 3, 4));
+        assert.isTrue(p1True.calledWith(1));
+        assert.isTrue(p2True.calledWith(2));
+        assert.isTrue(p3False.calledWith(3));
+        assert.isTrue(c2False.calledWith([true, true, false, 4]));
+        assert.isTrue(p1True.calledOnce);
+        assert.isTrue(p2True.calledOnce);
+        assert.isTrue(p3False.calledOnce);
+        assert.isTrue(c1False.calledOnce);
+        assert.isTrue(c2False.calledOnce);
+      });
+    });
+  });
+
+  context('with `all` as the combinator', function() {
+    context('with all predicates passing', function() {
+      specify('returns true', function() {
+        const f = argsPass(all, [p1True, p2True, p3True]);
+        assert.isTrue(f(1, 2, 3));
+      });
+    });
+
+    context('with a predicate failing', function() {
+      specify('returns true', function() {
+        const f = argsPass(all, [p1True, p2True, p3False]);
+        assert.isFalse(f(1, 2, 3));
+      });
+    });
+  });
+
+  context('with `any` as the combinator', function() {
+    context('with all predicates passing', function() {
+      specify('returns true', function() {
+        const f = argsPass(any, [p1True, p2True, p3True]);
+        assert.isTrue(f(1, 2, 3));
+      });
+    });
+
+    context('with one predicates passing', function() {
+      specify('returns true', function() {
+        const f = argsPass(any, [p1True, p2False, p3True]);
+        assert.isTrue(f(1, 2, 3));
+      });
+    });
+
+    context('with no predicate passing', function() {
+      specify('returns true', function() {
+        const f = argsPass(any, [p1False, p2False, p3False]);
+        assert.isFalse(f(1, 2, 3));
+      });
+    });
+  });
```

Comment:
please add `with 'none' as the combinator` tests

Reply:
Yes, we should test our own code examples.